mbed-hal/spi_api.h

```diff
@@ -52,10 +75,30 @@ typedef struct spi_s spi_t;
 
 #endif
 
+typedef enum {
+    SPI_RC_OK = 0,                          /**< No error occured */
+
+    SPI_RC_NOT_ACTIVE = 0,                  /**< Peripheral is not active */
+    SPI_RC_NOT_BUSY = 0,                    /**< Peripheral is not busy with a transfer */
+
+    SPI_RC_UNKNOWN_ERROR = -1,              /**< An unspecified error occurred */
+
+    SPI_RC_UNSUPPORTED_NUMBER_OF_BITS = -2, /**< Unsupported number of bits */
+    SPI_RC_UNSUPPORTED_MODE = -3,           /**< Unsupported spi mode */
+    SPI_RC_UNSUPPORTED_BIT_ORDER = -4,      /**< Unsupported spi bit order */
+
+    SPI_RC_TRANSFER_LENGTH_TOO_LARGE = -5,  /**< Either tx or rx length is too large to transmit */
+
+    SPI_RC_ACTIVE = -6,                     /**< Peripheral is active */
+    SPI_RC_BUSY = -6,                       /**< Peripheral is busy with a transfer */
```

Comment:
If you want to duplicate a value, make it explicit.

```C
SPI_RC_BUSY = SPI_RC_ACTIVE,
```
Then, there is no question about what you meant to do.

Also, this is why I hate negative return codes when combined with enums.  Can we please figure out how to use positive enums for return codes?